### PR TITLE
Automatically configure Sentry as a logger.

### LINKED ISSUE #19: logger.error() doesn't get logged into Sentry

### PR DESCRIPTION
**Pull request by Natim**
Fixes #19

### CODE REVIEW

**Comment by underyx** [comment]
So @Natim, thanks a lot for taking the time to do this!

Now, I worry a bit about implicitly changing the behavior of logging of all users (as I started writing here https://github.com/underyx/aiohttp-sentry/issues/19#issuecomment-509693378) — I can imagine there would be users who already have lots of `log.error` spam in the console already, and they're intentionally not sending those lines to Sentry to not flood their instance/account. So the bad news is, I think it'd be irresponsible for us to change behavior like this in a new release.

Worry not though, there's good news as well 😄 I can see this working as a configurable option. If we allow people to set `patch_logging=True` to enable this behavior, and we default that to `False`, I think that could be useful for our users *and* not betray anyone's existing expectations.

So now the only question is: would you as a user like this solution? I can't decide myself if it's better to have to find out about `patch_logging=True` from this project's documentation, or if at that point you don't care and might as well configure `raven` yourself.

Could you let me know which way you would've preferred as a user before you opened the issue?

**Comment by Natim** [comment]
I don't see why you would use Sentry and not care about logging errors there, it seems like the core of Sentry to log errors? Am I missing the point there ?

The logging level let you define if you want to log INFO, WARNING or ERROR but I don't see any project were you would use Sentry without logging the ERROR log level.

Does being able to configure the log_level seems like an acceptable switch?

**Comment by underyx** [comment]
Sadly, I have seen company-internal projects that don't take `log.error` too seriously and are using it very liberally compared to Sentry logging.

Furthermore, and this is an even larger problem perhaps: just instantiating the middleware doesn't mean that the user wants it to be active on every request of the app. Imagine a setup like this:

```python
dev_middleware = [SentryMiddleware()]
prod_middleware = []
app = web.Application(middlewares=dev_middleware if setting.DEBUG else prod_middleware)
```

Now, unexpectedly, Sentry reporting would be active even in production, even though the user is explicitly trying to disable that!

**Comment by Natim** [comment]
Your example shouldn't pass the review but I see your point.

```
middlewares = []
ff setting.DEBUG:
 middlewares.append(SentryMiddleware())

app = web.Application(middlewares=middleware)
```

If you are happy with a parameter, it seems like a good compromise to me.

**Comment by coulix** [comment]
> So @Natim, thanks a lot for taking the time to do this!
> 
> Now, I worry a bit about implicitly changing the behavior of logging of all users (as I started writing here [#19 (comment)](https://github.com/underyx/aiohttp-sentry/issues/19#issuecomment-509693378)) — I can imagine there would be users who already have lots of `log.error` spam in the console already, and they're intentionally not sending those lines to Sentry to not flood their instance/account. So the bad news is, I think it'd be irresponsible for us to change behavior like this in a new release.
> 
> Worry not though, there's good news as well 😄 I can see this working as a configurable option. If we allow people to set `patch_logging=True` to enable this behavior, and we default that to `False`, I think that could be useful for our users _and_ not betray anyone's existing expectations.
> 
> So now the only question is: would you as a user like this solution? I can't decide myself if it's better to have to find out about `patch_logging=True` from this project's documentation, or if at that point you don't care and might as well configure `raven` yourself.
> 
> Could you let me know which way you would've preferred as a user before you opened the issue?

What about `setup_sentry_logging` instead of `patch_logging` which seems too vague.

**Comment by underyx** [comment]
When reading `setup_sentry_logging`, I'd expect the parameter to maybe be aboult logging what the sentry is doing, instead of "sentrying what logging is doing", if that phrasing makes sense 😅 

I also feel like `patch` is the best verb to use, to make it clear that the parameter triggers a global side effect.

File: aiohttp_sentry/__init__.py [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
 import raven_aiohttp
+from raven.conf import setup_logging
+from raven.handlers.logging import SentryHandler
 
 
 @web.middleware
 class SentryMiddleware:
-    def __init__(self, sentry_kwargs=None, *, install_excepthook=True, loop=None):
+    def __init__(self, sentry_kwargs=None, *, install_excepthook=True, loop=None,
+                 patch_logging=False, sentry_log_level=logging.ERROR):
```
Comment (underyx) [inline aiohttp_sentry/__init__.py]: Thanks for making `sentry_log_level` configurable, I was gonna ask for that 🙃

**Comment by underyx** [comment]
So at this point, I'm ready to merge as soon as we have the feature documented! I'd add it in a subheading under https://github.com/underyx/aiohttp-sentry#configuration to explain this behavior and warn about the global side effect.

Let me know if you prefer to document it yourself, or would prefer me to do it while merging and releasing the new version (later today or tomorrow).

**Comment by Natim** [comment]
I tried something here: https://github.com/underyx/aiohttp-sentry/pull/20/commits/6c010492419b5e3aef750bdc0d8d56d510e73deb

But feel free to improve it :+1:

**Review by underyx — APPROVED** [review]
Awesome, thanks so much, especially for bearing with me through the discussion of the issue! This is ready to merge and I'll just make tiny amendments (like fixing `IF` -> `If` in the docs) — so you can sleep peacefully today, nothing else is needed from your side.

I'll aim to release 0.6.0 tomorrow.

Thank you again!

**Comment by underyx** [comment]
Thanks once again! Your release is now out at https://pypi.org/project/aiohttp-sentry/0.6.0/

I tested it a bit locally, but please check how it works in a real life project and let me know if it behaves as you expect, thank you :)

**Comment by Natim** [comment]
Wil do, thanks !

**Pull request closed without merging**